cookbooks/bcpc/recipes/cluster_local_repository.rb

```diff
@@ -51,18 +51,26 @@
 end
 
 unless node[:fqdn] == get_bootstrap
-  require 'tempfile'
-  bcpc_apt_key_path = Tempfile.new('bootstrap-gpg-key').path
+  ruby_block 'bcpc-apt-key-create' do
+    block do
+      require 'tempfile'
+      node.run_state['bcpc_apt_key_path'] =
+        Tempfile.new('bootstrap-gpg-key').path
```

Comment:
Ah thanks for pointing out the `:delete` later [down](https://github.com/http-418/chef-bach/blob/513cb474436dd48bf804f3283bacffe33ed6bf7c/cookbooks/bcpc/recipes/cluster_local_repository.rb#L87-L90)